lib/rex/proto/ms_dtyp.rb

```diff
@@ -23,6 +23,22 @@ class MsDtypAccessMask < BinData::Record
     bit2   :reserved1
     bit1   :ma
     bit1   :as
+    def bit_names
+      names = []
+      names << :GENERIC_READ if self.gr
+      names << :GENERIC_WRITE if self.gw
+      names << :GENERIC_EXECUTE if self.gx
+      names << :GENERIC_ALL if self.ga
+      names << :MAXIMUM_ALLOWED if self.ma
+      names << :ACCESS_SYSTEM_SECURITY if self.as
+      names << :SYNCHRONIZE if self.sy
+      names << :WRITE_OWNER if self.wo
+      names << :WRITE_DACL if self.wd
+      names << :READ_CONTROL if self.rc
+      names << :DELETE if self.de
+      names << :PROTOCOL_SPECIFIC if self.protocol
```

Comment:
This should be left out. It's protocol specific, so another protocol could inherit from this class and add it's own flags in here to extend the names.